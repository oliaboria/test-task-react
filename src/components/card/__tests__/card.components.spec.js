import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

import mockShow from '../../../mocks/show';
import Card from '../card.component';

describe('Card component', () => {
    it('should render card component with fallback image', () => {
        const mockShowWithoutImage = {
            ...mockShow,
            image: null,
        };

        const wrapper = shallow(<Card card={mockShowWithoutImage} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render card component with image', () => {
        const wrapper = shallow(<Card card={mockShow} />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
