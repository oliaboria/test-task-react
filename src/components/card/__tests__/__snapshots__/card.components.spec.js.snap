// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render card component with fallback image 1`] = `
<Fragment>
  <h1
    className="card-title"
  >
    The Powerpuff Girls
  </h1>
  <div
    className="clearfix"
  >
    <figure
      className="card-image left-column"
    >
      <img
        alt="Cover"
        src="no-img.png"
      />
    </figure>
    <div
      className="card-description right-column"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as <b>The Powerpuff Girls</b>. Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to \\"fight like a girl.\\"</p>",
        }
      }
    />
  </div>
</Fragment>
`;

exports[`Card component should render card component with image 1`] = `
<Fragment>
  <h1
    className="card-title"
  >
    The Powerpuff Girls
  </h1>
  <div
    className="clearfix"
  >
    <figure
      className="card-image left-column"
    >
      <img
        alt="Cover"
        src="http://static.tvmaze.com/uploads/images/medium_portrait/60/151357.jpg"
      />
    </figure>
    <div
      className="card-description right-column"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>The city of Townsville may be a beautiful, bustling metropolis, but don't be fooled! There's evil afoot! And only three things can keep the bad guys at bay: Blossom, Bubbles and Buttercup, three super-powered little girls, known to their fans (and villains everywhere) as <b>The Powerpuff Girls</b>. Juggling school, bedtimes, and beating up giant monsters may be daunting, but together the Powerpuff Girls are up to the task. Battling a who's who of evil, they show what it really means to \\"fight like a girl.\\"</p>",
        }
      }
    />
  </div>
</Fragment>
`;
